### PR TITLE
JPG-002 | User login Api

### PR DESCRIPTION
**Pull request by Bhavesh-SP18**

### CODE REVIEW

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 1]
```diff
@@ -37,4 +39,24 @@ public SignUpResponse signUp(@NonNull final SignUpRequest signUpRequest) throws
                 .message("Sign-up is successful.")
                 .build();
     }
+
+    public LoginResponse login (final LoginRequest loginRequest) throws Exception {
+        final Optional<UserEntity> byEmail =
```
Comment (kunalcodess) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:44]: Don't use variable names autogenerated by IDE, they mostly don't make sense

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 2]
```diff
@@ -37,4 +39,24 @@ public SignUpResponse signUp(@NonNull final SignUpRequest signUpRequest) throws
...
+                userRepository.findByEmail(loginRequest.getEmail());
+        if (byEmail.isPresent()) {
+            return LoginResponse.builder()
+                    .message(" user login successful")
+                    .build();
+        }
```
Comment (kunalcodess) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:50]: For now this is ok, put a TODO here for refactoring once you have exception handling layer. You should have a orElseThrow() to throw exception with a message

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 3]
```diff
@@ -37,4 +39,24 @@ public SignUpResponse signUp(@NonNull final SignUpRequest signUpRequest) throws
...
+        UserEntity save = userRepository.save(UserEntity.builder()
```
Comment (kunalcodess) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:51]: Why do you need save variable?

**Review by kunalcodess — CHANGES_REQUESTED** [review]
Why did we raise PR to main? This should be raised for master. Overall changes looks good, some minor fixes.

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 1 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java]: Okay sir, Next time will give our own name to variable :)

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 2 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java:72]: ok sir, ill try

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 1 above)
Comment (kunalcodess) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java]: Please correct these in next commit

**Review by kunalcodess — CHANGES_REQUESTED** [review]
Title of the PR must be 
JPG-002 | User login Api

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 3 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java]: sir, it was a by mistake :(

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,7 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/passbook
+spring.datasource.username=postgres
+spring.datasource.password=qwerty1201
+#spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
```
Comment (spark-coders) [inline src/main/resources/application.properties:6]: Read about dialect

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+        return SignUpResponse.builder()
+                .message("Sign-up is successful.")
+                .build();
+    }
+
+    public LoginResponse login(final LoginRequest loginRequest) throws Exception {
+
+
+        final Optional<UserEntity> byEmailAndPassword =
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:47]: Refactor  the below code

File: src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.Optional;
+
+public interface UserRepository  extends JpaRepository<UserEntity, Integer> {
+    Optional<UserEntity> findByEmail(String email);
+
+    Optional<UserEntity> findByEmailAndPassword(String email,String password);
+
+
+    Optional<UserEntity> findByPassword(String email);
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java:14]: Why is this required.

File: src/main/resources/application.properties (hunk 4 above)
Comment (Bhavesh-SP18) [reply src/main/resources/application.properties:6]: **- Dialect is a class and a bridge between Java JDBC types and SQL types, which contains mapping between java language data type and database datatype. 
- Dialect allows Hibernate to generate SQL optimized for a particular relational database.**

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 5 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java:47]: refactor done, but code not pushed yet.
